### ISSUE TITLE
Add support for CodeBuild with VPC

### ISSUE THREAD

**Issue by drsolaris** [terraform-providers/terraform-provider-aws#2495]
Amazon [announced](https://aws.amazon.com/blogs/devops/access-resources-in-a-vpc-from-aws-codebuild-builds/) support for running CodeBuild builds within a VPC last week.

### Affected Resource(s)

* aws_codebuild_project

### Expected Behavior

Create & update of projects should optionally allow the specification of VPC, subnets & security groups, likely within a new optional block to match the Create and Update APIs. Read will also need additions for these properties.

* If VPC is supplied, subnets must also be supplied.
* If subnets is supplied, VPC must also be supplied.
* If security_groups is supplied, both VPC and subnets must be supplied.

```hcl
resource "aws_codebuild_project" "foo" {
  name = "test-project"

  artifacts {
    type = "NO_ARTIFACTS"
  }

  environment {
    compute_type = "BUILD_GENERAL1_SMALL"
    image        = "2"
    type         = "LINUX_CONTAINER"
  }

  source {
    type     = "GITHUB"
    location = "https://github.com/mitchellh/packer.git"
  }

  vpc_config {
    vpc_id = "vpc-725fca"

    subnets = [
       "subnet-ba35d2e0",
       "subnet-ab129af1",
    ]

    security_group_ids = [
       "sg-f9f27d91",
       "sg-e4f48g23",
    ]
  }
}
```

### References

[Blog post](https://aws.amazon.com/blogs/devops/access-resources-in-a-vpc-from-aws-codebuild-builds/)
[Documentation](https://docs.aws.amazon.com/codebuild/latest/userguide/vpc-support.html)

**Comment by drsolaris** [comment]
Would be my first time making a contribution to a Go project, but I might pick this up.

Had a look through the API documentation for [CreateProject](https://docs.aws.amazon.com/codebuild/latest/APIReference/API_CreateProject.html) and it seems VPC support was added as a new nested JSON section, so I've update the above proposed HCL to match.

**Comment by kilsbo** [comment]
I'm really looking for this as well. Only way (AWS is talking about) to access RDS instances from your continuous deployment.

**Comment by bflad** [comment]
This functionality has been merged into master via #2547 and #3324. It will be released in v1.10.0 of the AWS provider, likely tomorrow. 🎉

**Comment by bflad** [comment]
This has been released in [version 1.10.0 of the AWS provider](https://github.com/terraform-providers/terraform-provider-aws/blob/master/CHANGELOG.md#1100-february-24-2018). Please see the [Terraform documentation on provider versioning](https://www.terraform.io/docs/configuration/providers.html#provider-versions) or reach out if you need any assistance upgrading.
